duration.go

```diff
@@ -0,0 +1,21 @@
+package claircore
+
+import "time"
+
+// Duration is a serializeable [time.Duration].
+type Duration time.Duration
+
+// UnmarshalText implements [encoding.TextUnmarshaler].
+func (d *Duration) UnmarshalText(b []byte) error {
+	dur, err := time.ParseDuration(string(b))
+	if err != nil {
+		return err
+	}
+	*d = Duration(dur)
+	return nil
+}
+
+// MarshalText implements [encoding.TextMarshaler].
+func (d *Duration) MarshalText() ([]byte, error) {
+	return []byte(time.Duration(*d).String()), nil
```

Comment:
do we want to consider a `nil` check here?

Reply:
I think it'd be pretty difficult to arrive there (particularly with the current way of instantiating) but possible no the less.